testing/test_pathlib.py

```diff
@@ -569,6 +570,29 @@ def test_samefile_false_negatives(tmp_path: Path, monkeypatch: MonkeyPatch) -> N
     assert getattr(module, "foo")() == 42
 
 
+def test_scandir_with_non_existent_directory() -> None:
+    # Test with a directory that does not exist
+    non_existent_dir = "path_to_non_existent_dir"
+    result = scandir(non_existent_dir)
+    # Assert that the result is an empty list
+    assert result == []
+
+
+def test_scandir_handles_os_error():
```

Comment:
```suggestion
def test_scandir_handles_os_error() -> None:
```

Reply:
I'm not sure we need this test actually...

Reply:
> I'm not sure we need this test actually... 

This test is essential for achieving full coverage. Without it, the raise command in the scandir function will not be covered, resulting in a 90% coverage score.